src/microsoft-dotnet-interactive/src/javascriptKernel.ts

```diff
@@ -11,15 +11,34 @@ export class JavascriptKernel extends Kernel {
     private capture: ConsoleCapture;
 
     constructor(name?: string) {
-        super(name ?? "javascript", "Javascript");
+        super(name ?? "javascript", "JavaScript");
         this.suppressedLocals = new Set<string>(this.allLocalVariableNames());
         this.registerCommandHandler({ commandType: contracts.SubmitCodeType, handle: invocation => this.handleSubmitCode(invocation) });
         this.registerCommandHandler({ commandType: contracts.RequestValueInfosType, handle: invocation => this.handleRequestValueInfos(invocation) });
         this.registerCommandHandler({ commandType: contracts.RequestValueType, handle: invocation => this.handleRequestValue(invocation) });
+        this.registerCommandHandler({ commandType: contracts.SendValueType, handle: invocation => this.handleSendValue(invocation) });
 
         this.capture = new ConsoleCapture();
     }
 
+    private handleSendValue(invocation: IKernelCommandInvocation): Promise<void> {
+        const sendValue = <contracts.SendValue>invocation.commandEnvelope.command;
+        if (sendValue.formattedValue) {
+            switch (sendValue.formattedValue.mimeType) {
+                case 'application/json':
+                    (<any>globalThis)[sendValue.name] = JSON.parse(sendValue.formattedValue.value);
+                    break;
+                case 'text/plain':
```

Comment:
This MIME type feels wrong to me. Maybe `text/javascript`?

Reply:
if the variable is shared as plain/text we trat that as string value. what should be the behaviour ?

Reply:
I'm not sure this MIME type should be valid, or else we should just declare a string.

Reply:
right now if is text/plain it declares a string.

Reply:
Deleted that case